apps/mobile/src/features/settings/app-authentication-sheet.tsx

```diff
@@ -1,19 +1,41 @@
-import { RefObject } from 'react';
+import { RefObject, useEffect } from 'react';
 
-import { useSettings } from '@/store/settings/settings.write';
+import { useToastContext } from '@/components/toast/toast-context';
+import { useSettings } from '@/store/settings/settings';
 import { t } from '@lingui/macro';
 
-import { Sheet, SheetRef, Text } from '@leather.io/ui/native';
+import { ItemLayout, KeyholeIcon, SheetRef, Switch, TouchableOpacity } from '@leather.io/ui/native';
+
+import { SettingsSheetLayout } from './settings-sheet.layout';
 
 interface AppAuthenticationSheetProps {
   sheetRef: RefObject<SheetRef>;
 }
 export function AppAuthenticationSheet({ sheetRef }: AppAuthenticationSheetProps) {
-  const { theme: themeVariant } = useSettings();
+  const settings = useSettings();
+  const { displayToast } = useToastContext();
+
+  useEffect(() => {
+    if (settings.securityLevelPreference === 'undefined')
+      settings.changeSecurityLevelPreference('insecure');
+  });
```

Comment:
Mmm, this seems weird? Why wouldn't we just set the preference as `insecure` in the store if it it changed automatically when landing on this page?

Reply:
Yeah, I'll remove. Prob not needed but was asking for clarification on this setting from Edgar. I'm not sure why undefined is even needed in the type?